src/cargo/util/lints.rs

```diff
@@ -605,6 +608,164 @@ fn output_unknown_lints(
     Ok(())
 }
 
+const UNEXPECTED_CFGS: Lint = Lint {
+    name: "unexpected_cfgs",
+    desc: "lint on unexpected target cfgs",
+    groups: &[],
+    default_level: LintLevel::Warn,
+    edition_lint_opts: None,
+    feature_gate: None,
```

Comment:
Right now only the acquisition of `check-cfg`s from the compiler is feature gated.  The lint should be feature gated as well.